module/VuFind/src/VuFind/ChannelProvider/ListItems.php

```diff
@@ -282,22 +260,17 @@ protected function getLists()
     protected function getListsByTagAndId()
     {
         // Get public lists by search criteria
-        $lists = $this->tagService->getListsForTag(
+        $listIds = $this->tagService->getListsForTag(
             $this->tags,
             $this->ids,
             true,
             $this->andTags
         );
 
-        // Format result set into an array:
         $result = [];
-        if (count($lists)) {
-            $callback = function ($select) use ($lists) {
-                $select->where->in('id', $lists);
-            };
-
-            foreach ($this->userList->select($callback) as $list) {
-                $result[] = $list;
+        if (count($listIds)) {
+            foreach ($listIds as $id) {
+                $result[] = $this->listService->getEntityById(\VuFind\Db\Entity\UserList::class, $id);
```

Comment:
(It actually looks like there are two places in this class where we iterate to fetch lists -- in getListsById and in getListsByTagAndId. Seems like it would be worthwhile to change both to use a new method that accepts an array of IDs and returns an array of entity objects, since I assume there's a way to do that with fewer queries. (And even if there isn't, abstracting the process would offer an opportunity for future optimization).